File: liqp-core/src/main/java/liqp/nodes/EqNode.java
package liqp.nodes;

import liqp.node.LNode;
import liqp.node.LValue;
import lombok.Getter;

@Getter
public class EqNode extends ExpressionNode{

  public EqNode(LNode lhs, LNode rhs) {
    super(lhs, rhs);
  }

  @Override
    public Object render(LContext context) {

        Object a = lhs.render(context);
        Object b = rhs.render(context);

        return LValue.Companion.areEqual(a, b);

    }
}

File: liqp-core/src/main/kotlin/liqp/nodes/EqNode.kt
package liqp.nodes

import liqp.ComparisonResult
import liqp.context.LContext
import liqp.node.LNode

class EqNode(lhs: LNode, rhs: LNode) : ExpressionNode(lhs, rhs) {

  override fun render(context: LContext): Any {
    val a = lhs.render(context)
    val b = rhs.render(context)
    return context.areEqual(a, b)
  }
}
